indexer/src/main/java/au/org/aodn/esindexer/model/AssociateRecordFactory.java

```diff
@@ -0,0 +1,84 @@
+package au.org.aodn.esindexer.model;
+
+import au.org.aodn.stac.model.LinkModel;
+import au.org.aodn.stac.util.JsonUtil;
+
+import java.util.*;
+
+import static au.org.aodn.esindexer.utils.CommonUtils.safeGet;
+
+public class AssociateRecordFactory {
+
+    private record RecordTitle(String recordTitle, String recordAbstract) {}
+
+    public static List<LinkModel> buildAssociatedRecords(Map<String, ?> data) {
+        var records = new ArrayList<LinkModel>();
+        if (data == null || data.isEmpty()) {
+            return records;
+        }
+        var parentsRecordMap = getRecordMaps(data, "parent");
+        if (!parentsRecordMap.isEmpty()) {
+            var parentLink = buildLink(parentsRecordMap.get(0), RelationType.PARENT);
+            if (parentLink != null) {
+                records.add(parentLink);
+            }
+        }
+        var siblingRecordMaps = getRecordMaps(data, "brothersAndSisters");
+        var childRecordMaps = getRecordMaps(data, "children");
+
+        var siblingLinks = buildLinks(siblingRecordMaps, RelationType.SIBLING);
+        if (!siblingLinks.isEmpty()) {
+            records.addAll(siblingLinks);
+        }
+
+        var childLinks = buildLinks(childRecordMaps, RelationType.CHILD);
+        if (!childLinks.isEmpty()) {
+            records.addAll(childLinks);
+        }
```

Comment:
should consider renaming this block, something be like:

```java
var siblingRecordsLinks = getLinksFromMultiRecordsByRelationType(siblingRecords, RelationType.SIBLING);
if (!siblingRecordsLinks.isEmpty()) {
    records.addAll(siblingRecordsLinks);
}

var childRecordsLinks = getLinksFromMultiRecordsByRelationType(childRecords, RelationType.CHILD);
if (!childRecordsLinks.isEmpty()) {
    records.addAll(childRecordsLinks);
}
```

Reply:
i refactored this part to be clearer